tests/Behat/Context/Application/RefundingContext.php

```diff
@@ -115,8 +116,27 @@ public function decideToRefundPartFromProductWithPayment(
     public function decideToRefundOrderShipment(PaymentMethodInterface $paymentMethod): void
     {
         $shippingAdjustment = $this->order->getAdjustments(AdjustmentInterface::SHIPPING_ADJUSTMENT)->first();
+        $remainingTotal = $this->remainingTotalProvider->getTotalLeftToRefund($shippingAdjustment->getId(), RefundType::shipment());
 
-        $this->commandBus->dispatch(new RefundUnits($this->order->getNumber(), [], [$shippingAdjustment->getId()], $paymentMethod->getId(), ''));
+        $this->commandBus->dispatch(new RefundUnits(
+            $this->order->getNumber(), [], [new ShipmentRefund($shippingAdjustment->getId(), $remainingTotal)], $paymentMethod->getId(), ''
+        ));
+    }
+
+    /**
+     * @When /^I decide to refund ("[^"]+") from order shipment with ("[^"]+" payment)$/
+     */
+    public function decideToRefundPartOfOrderShipment(int $amount, PaymentMethodInterface $paymentMethod): void
+    {
+        $shippingAdjustment = $this->order->getAdjustments(AdjustmentInterface::SHIPPING_ADJUSTMENT)->first();
+
+        try {
+            $this->commandBus->dispatch(new RefundUnits(
+                $this->order->getNumber(), [], [new ShipmentRefund($shippingAdjustment->getId(), $amount)], $paymentMethod->getId(), ''
+            ));
+        } catch (CommandDispatchException $exception) {
+            return;
```

Comment:
```throw new \Exception('RefundUnits command should not fail');```

wdyt?

Reply:
I this specific case we want to silence the error, to check if anything changed later... but I think this step could be named `I try to refund` instead of `I decide to refund`, to avoid confusion... wdyt?